### PR TITLE
Add weight prop to Heading

### LINKED ISSUE #6276: Heading — support `weight` and/or `kind`
### Expected Behavior

It is desirable to be able to control the heading weight -- sometimes bold is desired, other times a lighter weight is desired. The caller should be able to control that directly by `weight` (like text) and/or something that could be defined as a `kind` in theme (like Button theming) depending on level of flexibility desired.

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by halocline**
#### What does this PR do?

Closes #6276 by adding `weight` to `<Heading />`

#### Where should the reviewer start?

Heading.js

#### What testing has been done on this PR?

Jest test suite, Storybook, plus new snapshot.

#### How should this be manually tested?

```
<Heading weight="normal">My Heading</Heading>
<Heading weight="bold">My Heading</Heading>
<Heading weight={700}>My Heading</Heading>
```

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [x] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?

#6276

#### Do the grommet docs need to be updated?

Yes.

#### Should this PR be mentioned in the release notes?

Yes.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

File: src/js/components/Heading/StyledHeading.js [hunk 1]
```diff
@@ -20,9 +20,11 @@ const sizeStyle = props => {
         font-size: ${data ? data.size : size};
         line-height: ${data ? data.height : 'normal'};
         max-width: ${(props.fillProp && 'none') ||
-          (data && data.maxWidth) ||
-          levelStyle.medium.maxWidth};
-        font-weight: ${levelStyle.font.weight || headingTheme.weight};
+        (data && data.maxWidth) ||
+        levelStyle.medium.maxWidth};
```
Comment (MikeKingdom) [inline src/js/components/Heading/StyledHeading.js:24]: could use the indent. It's a little hard to read which styles are involved

File: src/js/components/Heading/StyledHeading.js [hunk 2]
```diff
@@ -20,9 +20,11 @@ const sizeStyle = props => {
...
+        font-weight: ${props.weight ||
+        levelStyle.font.weight ||
+        headingTheme.weight};
```
Comment (MikeKingdom) [inline src/js/components/Heading/StyledHeading.js:27]: ditto

**Review by MikeKingdom — COMMENTED** [review]
Looks great. Minor indent comment...

File: src/js/components/Heading/StyledHeading.js (hunk 1 above)
Comment (halocline) [reply src/js/components/Heading/StyledHeading.js:24]: hmmm... let me see what I can do. I believe Prettier was forcing this.

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/components/Heading/StyledHeading.js (hunk 1 above)
Comment (halocline) [reply src/js/components/Heading/StyledHeading.js:24]: Something in the Prettier/ESLint space is forcing this onSave. I couldn't track down the rule / plugin causing this in a reasonable time. Okay with proceeding as is?

**Review by jcfilben — APPROVED** [review]
Looks good! We will need a PR on grommet-site to update the docs

**Pull request merged**